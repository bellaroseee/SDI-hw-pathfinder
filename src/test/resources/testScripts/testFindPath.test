# Test findPath 1 hop

CreateGraph graph1

AddNode graph1 PointA
AddNode graph1 PointB

AddEdge graph1 PointA PointB 10.0
AddEdge graph1 PointA PointB 5.0

FindPath graph1 PointA PointB

# Test findPath 2 hops

AddNode graph1 PointC

AddEdge graph1 PointB PointC 2.0
AddEdge graph1 PointA PointC 15.0

FindPath graph1 PointA PointC

# Test findPath cyclic

AddEdge graph1 PointC PointA 1.0

FindPath graph1 PointA PointA