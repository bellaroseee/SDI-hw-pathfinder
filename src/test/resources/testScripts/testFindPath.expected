# Test findPath 1 hop

created graph graph1

added node PointA to graph1
added node PointB to graph1

added edge 10.000 from PointA to PointB in graph1
added edge 5.000 from PointA to PointB in graph1

path from PointA to PointB:
PointA to PointB with weight 5.000
total cost: 5.000

# Test findPath 2 hops

added node PointC to graph1

added edge 2.000 from PointB to PointC in graph1
added edge 15.000 from PointA to PointC in graph1

path from PointA to PointC:
PointA to PointB with weight 5.000
PointB to PointC with weight 2.000
total cost: 7.000

# Test findPath cyclic

added edge 1.000 from PointC to PointA in graph1

path from PointA to PointA:
total cost: 0.000